pool/web/handler/admin_contacts.ml

```diff
@@ -10,29 +10,35 @@ let experiment_id = HttpUtils.find_id Experiment.Id.of_string Field.Experiment
 
 let index req =
   let open Utils.Lwt_result.Infix in
-  let result ({ Pool_context.database_label; user; _ } as context) =
-    Utils.Lwt_result.map_error (fun err -> err, "/admin/dashboard")
-    @@
-    let query =
-      let open Contact in
-      Query.from_request ~default:default_query ~searchable_by ~sortable_by req
-    in
-    let* actor =
-      Pool_context.Utils.find_authorizable ~admin_only:true database_label user
-    in
-    let%lwt contacts =
-      Contact.find_all
-        ~query
-        ~actor
-        ~permission:Contact.Guard.Access.index_permission
-        database_label
-        ()
-    in
-    Page.Admin.Contact.index context contacts
-    |> create_layout req ~active_navigation:"/admin/contacts" context
-    >|+ Sihl.Web.Response.of_html
+  let active_navigation = "/admin/contacts" in
+  let error_path = "/admin/dashboard" in
+  extract_happy_path req
+  @@ fun ({ Pool_context.database_label; user; _ } as context) ->
+  let query =
+    let open Contact in
+    Query.from_request ~default:default_query ~searchable_by ~sortable_by req
   in
-  result |> extract_happy_path req
+  let* actor =
+    Pool_context.Utils.find_authorizable ~admin_only:true database_label user
+    >|- fun err -> err, error_path
+  in
+  let%lwt contacts, query =
+    Contact.find_all
+      ~query
+      ~actor
+      ~permission:Contact.Guard.Access.index_permission
+      database_label
+      ()
+  in
+  let page = Page.Admin.Contact.index context contacts query in
+  if Htmx.is_hx_request req
+  then Ok (HttpUtils.Htmx.html_to_plain_text_response page) |> Lwt_result.lift
+  else
+    let* view =
+      create_layout ~active_navigation req context page
+      >|- fun err -> err, error_path
+    in
+    Ok (Sihl.Web.Response.of_html view) |> Lwt_result.lift
```

Comment:
The following code sequence is the same for all changed pages, i'd suggest adding it to HttpUtils

```
  if Htmx.is_hx_request req
  then Ok (HttpUtils.Htmx.html_to_plain_text_response page) |> Lwt_result.lift
  else
    let* view =
      create_layout ~active_navigation req context page
      >|- fun err -> err, error_path
    in
    Ok (Sihl.Web.Response.of_html view) |> Lwt_result.lift
```

And instead of `Ok (...) |> Lwt_result.lift` use directly `... |> Lwt_result.return`

Reply:
Yes! Now that we have a clear pattern I think we can abstract it away :) 

I was considering something like this:

```ocaml
let index = HttpUtils.htmx_handler ~with_query:(module Filter) @@ fun ctx query ->
  let%lwt filter_list, query = Filter.find_by query database_label in
  let page = Page.Admin.Filter.index context filter_list query in
  Lwt_result.return page
```

but we could also just do the query manually on every handler:

```ocaml
let index = HttpUtils.htmx_handler @@ fun ctx ->
  let query =
    Query.from_request
      ~searchable_by:Filter.searchable_by
      ~sortable_by:Filter.sortable_by
      ~default:Filter.default_query
      req
  in
  let%lwt filter_list, query = Filter.find_by query database_label in
  let page = Page.Admin.Filter.index context filter_list query in
  Lwt_result.return page
```

wdyt?

Reply:
I'd prefer the first approach, where the `(module Filter)` exposes e.g. the "searchable"/"sortable" functions.

Looks good to me!